Source/Shared/arcana/threading/task.h

```diff
@@ -111,9 +111,18 @@ namespace arcana
                     })
             ) };
 
-            m_payload->create_continuation([&scheduler](auto&& c)
+            m_payload->create_continuation([this, &scheduler, token](auto&& c) mutable
```

Comment:
Spoke about this fairly extensively offline. `this` is probably not a good thing to capture, but there are type erasure problems with passing it as an argument, so an alternative is to capture `m_payload` directly. `token` does, I think, need to be captured here because (1) it's guarding the scheduler, which is captured here, and (2) the `continuation_payload` which calls the scheduling lambda doesn't know about cancellation, so it doesn't have it to pass in.